### DISCUSSION TITLE
I received high severity message: Path traversal in webpack-dev-middleware

### DISCUSSION THREAD

**Discussion by ginarusso** [community/community#114668] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

**I'm not sure what to do - thank you for any suggestions. Please see below:**
[https://github.com/advisories/GHSA-wr3j-pwj9-hqq6]
**Summary**
The webpack-dev-middleware middleware does not validate the supplied URL address sufficiently before returning the local file. It is possible to access any file on the developer's machine
**Details**
The middleware can either work with the physical filesystem when reading the files or it can use a virtualized in-memory memfs filesystem.
**Impact**
The developers using webpack-dev-server or webpack-dev-middleware are affected by the issue. When the project is started, an attacker might access any file on the developer's machine and exfiltrate the content (e.g. password, configuration files, private source code, ...).

If the development server is listening on a public IP address (or 0.0.0.0), an attacker on the local network can access the local files without any interaction from the victim (direct connection to the port).

If the server allows access from third-party domains (CORS, Allow-Access-Origin: * ), an attacker can send a malicious link to the victim. When visited, the client side script can connect to the local server and exfiltrate the local files.

**Recommendation**
The URL should be unescaped and normalized before any further processing.

**Comment by lenichols** [comment]
+1

**Comment by ginarusso** [comment]
Any help would be appreciated.
